src/UPSRateRequest.php

```diff
@@ -134,18 +146,30 @@ public function getRates(ShipmentInterface $commerce_shipment, ShippingMethodInt
 
         $price = new Price((string) $cost, $currency);
         $service_name = $ups_rate->Service->getName();
+        $date = new DrupalDateTime();
+        $date->format('Y-m-d');
 
         $shipping_service = new ShippingService(
           $service_code,
           $service_name
         );
-        $rates[] = new ShippingRate(
-          $service_code,
-          $shipping_service,
-          $price
+
+        $times = $this->upsTransit->getTransitTime(
+          $this->configuration,
+          $commerce_shipment,
+          $shipment
         );
+        foreach ($times->ServiceSummary as $serviceSummary) {
+          $rates[] = new ShippingRate(
+            $service_code,
+            $shipping_service,
+            $price,
+            $date::createFromFormat('Y-m-d', $serviceSummary->EstimatedArrival->getDate())
+          );
+        }
       }
     }
+
```

Comment:
**CHANGE REQUEST**

Can we do the usual trick here to improve readability?

```
if (empty($ups_rates->RatedShipment)) {
  return $rates;
}
```

Then if you can refactor the logic in the foreach loop into a separate function, so that it's something like:

```
foreach ($ups_rates->RatedShipment as $ups_rate) {
  $service_code = $ups_rate->Service->getCode();

  // Only add the rate if this service is enabled.
  if (!in_array($service_code, $this->configuration['services'])) {
    continue;
  }

  $rates[] = $this->someFunction($ups_rate);
}

return $rates;
```